components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java

```diff
@@ -44,39 +56,93 @@ public void init() throws ArtifactSynchronizerException {
         try {
-            ByteArrayInputStream byteStream =
-                    apiMgtDAO.getGatewayPublishedAPIArtifacts(APIId, gatewayLabel, gatewayInstruction);
-            byte[] bytes = ByteStreams.toByteArray(byteStream);
-            gatewayRuntimeArtifacts = new String(bytes);
-            if (log.isDebugEnabled()) {
-                log.debug("Successfully retrieved Artifact of " + APIId);
-            }
-        } catch (APIManagementException | IOException e) {
-            throw new ArtifactSynchronizerException("Error retrieving Artifact belongs to  " + APIId + " from DB", e);
+            String baseURL = APIConstants.HTTPS_PROTOCOL_URL_PREFIX +
+                    System.getProperty(APIConstants.KEYMANAGER_HOSTNAME) + ":" +
+                    System.getProperty(APIConstants.KEYMANAGER_PORT) + APIConstants.INTERNAL_WEB_APP_EP;
+            String endcodedgatewayLabel= URLEncoder.encode(gatewayLabel, APIConstants.DigestAuthConstants.CHARSET);
+            String path  = APIConstants.GatewayArtifactSynchronizer.SYNAPSE_ARTIFACTS + "?apiId=" + APIId +
+                    "&gatewayInstruction=" + gatewayInstruction +"&gatewayLabel="+ endcodedgatewayLabel;
+            String endpoint = baseURL + path;
+            HttpResponse httpResponse = invokeService(endpoint);
+            return EntityUtils.toString(httpResponse.getEntity(), APIConstants.DigestAuthConstants.CHARSET);
+        } catch (IOException e) {
+            String msg = "Error while executing the http client " ;
+            log.error(msg, e);
+            throw new ArtifactSynchronizerException(msg, e);
         }
-        return gatewayRuntimeArtifacts;
     }
 
     @Override
     public List<String> retrieveAllArtifacts(String label) throws ArtifactSynchronizerException {
         List<String> gatewayRuntimeArtifactsArray = new ArrayList<>();
         try {
-            List<ByteArrayInputStream> baip = apiMgtDAO.getAllGatewayPublishedAPIArtifacts(label);
-            for (ByteArrayInputStream byteStream :baip){
-                byte[] bytes = ByteStreams.toByteArray(byteStream);
-                String  gatewayRuntimeArtifacts = new String(bytes);
-                gatewayRuntimeArtifactsArray.add(gatewayRuntimeArtifacts);
+            String baseURL = APIConstants.HTTPS_PROTOCOL_URL_PREFIX + System.getProperty(APIConstants.KEYMANAGER_HOSTNAME) + ":" +
+                    System.getProperty(APIConstants.KEYMANAGER_PORT) + APIConstants.INTERNAL_WEB_APP_EP;
+            String endcodedgatewayLabel= URLEncoder.encode(label, APIConstants.DigestAuthConstants.CHARSET);
+            String path  = APIConstants.GatewayArtifactSynchronizer.GATEAY_SYNAPSE_ARTIFACTS
+                    + "?gatewayLabel="+ endcodedgatewayLabel;
+            String endpoint = baseURL + path;
+            HttpResponse httpResponse = invokeService(endpoint);
+            String responseString = EntityUtils.toString(httpResponse.getEntity(), APIConstants.DigestAuthConstants.CHARSET);
```

Comment:
Better to do a null check before retrieving the entity

Reply:
https://github.com/wso2/carbon-apimgt/commit/0007abb08c126e094b757b72e03eba846516e404